x/ccv/provider/keeper/relay_test.go

```diff
@@ -402,85 +403,118 @@ func TestHandleSlashPacket(t *testing.T) {
+						providerConsAddr).Return(true).Times(1),
 				}
 			},
+			0,
 		},
 		{
 			"drop packet when infraction height not found",
-			ccv.SlashPacketData{ValsetUpdateId: 78}, // Keeper doesn't have a height mapped to this vscID.
+			ccv.SlashPacketData{
+				Validator:      tmtypes.Validator{Address: consumerConsAddr},
+				ValsetUpdateId: 78, // Keeper doesn't have a height mapped to this vscID.
+				Infraction:     stakingtypes.Downtime,
+			},
+
 			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
 				expectedPacketData ccv.SlashPacketData,
 			) []*gomock.Call {
 				return []*gomock.Call{
 
 					mocks.MockStakingKeeper.EXPECT().GetValidatorByConsAddr(
-						ctx, sdk.ConsAddress(expectedPacketData.Validator.Address)).Return(
+						ctx, providerConsAddr).Return(
 						stakingtypes.Validator{}, true,
 					).Times(1),
 
 					mocks.MockSlashingKeeper.EXPECT().IsTombstoned(ctx,
-						sdk.ConsAddress(expectedPacketData.Validator.Address)).Return(false).Times(1),
+						providerConsAddr).Return(false).Times(1),
 				}
 			},
+			0,
 		},
 		{
 			"full downtime packet handling, uses init chain height and non-jailed validator",
-			*ccv.NewSlashPacketData(abci.Validator{}, 0, stakingtypes.Downtime), // ValsetUpdateId = 0 uses init chain height.
+			*ccv.NewSlashPacketData(
+				tmtypes.Validator{Address: consumerConsAddr},
+				0, // ValsetUpdateId = 0 uses init chain height.
+				stakingtypes.Downtime),
 			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
 				expectedPacketData ccv.SlashPacketData,
 			) []*gomock.Call {
 				return testkeeper.GetMocksForHandleSlashPacket(
-					ctx, mocks, expectedPacketData, stakingtypes.Validator{Jailed: false},
-					true) // expectJailing = true
+					ctx, mocks,
+					providerConsAddr,                      // expected provider cons addr returned from GetProviderAddrFromConsumerAddr
+					stakingtypes.Validator{Jailed: false}, // staking keeper val to return
+					true)                                  // expectJailing = true
 			},
+			1,
 		},
 		{
 			"full downtime packet handling, uses valid vscID and jailed validator",
-			*ccv.NewSlashPacketData(abci.Validator{}, validVscID, stakingtypes.Downtime),
+			*ccv.NewSlashPacketData(
+				tmtypes.Validator{Address: consumerConsAddr},
+				validVscID,
+				stakingtypes.Downtime),
 			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
 				expectedPacketData ccv.SlashPacketData,
 			) []*gomock.Call {
 				return testkeeper.GetMocksForHandleSlashPacket(
-					ctx, mocks, expectedPacketData, stakingtypes.Validator{Jailed: true},
-					false) // expectJailing = false, validator is already jailed.
+					ctx, mocks,
+					providerConsAddr,                     // expected provider cons addr returned from GetProviderAddrFromConsumerAddr
+					stakingtypes.Validator{Jailed: true}, // staking keeper val to return
+					false)                                // expectJailing = false, validator is already jailed.
 			},
+			1,
```

Comment:
Can someone sanity check me that a slash ack should still be queued up even if the requested validator to jail is already jailed?

Reply:
Yes that's correct. Otherwise, a such validator would never have its slash flag reset on the consumer.